### DISCUSSION TITLE
File extension support nomination

### DISCUSSION THREAD

**Discussion by 0x7c13** [0x7c13/Notepads#819] [General]
Notepads currently only support a set of well-known file extensions (Check it out here: https://github.com/JasonStein/Notepads/blob/master/src/Notepads/Services/FileExtensionProvider.cs). If you are using Notepads for other types that are not supported yet, please feel free to let me know down below. Also, let me know what kind of file it is and why you think Notepads should support it.

Just remember: Notepads is a plain text editor for common use, not a code editor in anyway. So do not expect me to support all kinds of programming languages, especially those rarely used file extensions come with it.

Thanks!

**Comment by Bitwolfies** [comment]
.glslp (its different I swear)
.m3u (its a playlist file, but can be edited with plaintext)
.cht (retroarch cheatfile, its plaintext.)

**Comment by Bitwolfies** [comment]
.opt Retroarch core settings, plaintext.

**Comment by Teraskull** [comment]
> .m3u (its a playlist file, but can be edited with plaintext)

Also, .cue can be added.

.env (Environment file)
.toml (Configuration file, like .yaml or .ini)

**Comment by hello283** [comment]
加入.h文件

**Comment by 0x7c13** [comment]
> 加入.h文件

.h很早以前就支持了啊，而且一直都支持啊
![image](https://user-images.githubusercontent.com/1166162/83211505-1d395000-a112-11ea-94f4-8b514bbc3a22.png)
![image](https://user-images.githubusercontent.com/1166162/83211513-24605e00-a112-11ea-9e15-aba96ca583c2.png)

**Comment by Teraskull** [comment]
```.pot```, ```.po```, ```.mo``` - localization files used for i18n translations.

**Comment by soumyamahunt** [comment]
> `.pot`, `.po`, `.mo` - localization files used for i18n translations.

If you can, you should upgrade to Windows 10 2004. In 2004 and later versions you can open any kind of file (file with any or no extension) with notepads from the open with menu or set notepads as default for that kind of file.

**Comment by Teraskull** [comment]
> If you can, you should upgrade to Windows 10 2004.

Thanks, I'm aware, I am an insider myself, just thought that not everyone can/wants to upgrade to 2004.

**Comment by ghost** [comment]
.pem(SSL证书文件)
.key(密钥文件)

**Comment by updeshxp** [comment]
`.conf` , `.inf` , `.properties` , `.config` files used by various applications
